### PR TITLE
Modified CifParser.check() as one possible solution for issue #3626

### PR DESCRIPTION
**Pull request by kaueltzen**
## Summary

As mentioned in issue #3626 , `CifParser.check()` raises a `UserWarning: Incorrect stoichiometry` for a cif, if
- the keys `_chemical_formula_sum` and `_chemical_formula_structural` are missing and
- crystallographic sites have different multiplicities

For the relative stoichiometry to be checked correctly, the site multiplicities would have to be included by applying the cif's symmetry operations to the crystallographically unique sites (like for construction for the structure).

As this would be no longer a sanity check, i propose skipping the relative stoichiometry check in this case (but still including all the other checks for valid elements etc.).

For the case of skipping the relative stoichiometry check, i appended a message to `self.warnings`, but I'm not sure if there are better ways to handle this case.

### CODE REVIEW

**Comment by kaueltzen** [comment]
If you are okay with this approach i could adapt the 2 cif tests that are failing now (and for example add a formula key so the checks are not skipped).

File: pymatgen/io/cif.py [hunk 1]
```diff
@@ -1370,17 +1374,20 @@ def check(self, structure: Structure) -> str | None:
...
+                    for elt_a in orig_comp_elts
+                    for elt_b in orig_comp_elts
+                )
 
-            if not same_stoich:
-                failure_reason = f"Incorrect stoichiometry:\n  CIF={orig_comp}\n  PMG={struct_comp}\n  {ratios=}"
+                if not same_stoich:
+                    failure_reason = f"Incorrect stoichiometry:\n  CIF={orig_comp}\n  PMG={struct_comp}\n  {ratios=}"
+            else:
+                self.warnings.append("Skipping relative stoichiometry check because CIF does not contain formula keys.")
```
Comment (janosh) [inline pymatgen/io/cif.py:1390]: can we add a test for this warning?

**Comment by esoteric-ephemera** [comment]
Thanks @kaueltzen  for pointing this out! I like your solution of warning that the stoichiometry check couldn't be accomplished - the fallback I added originally clearly isn't robust enough in general

Adding a test for this (using the magnetic cif you sent with the issue) would be great to move this forward

**Comment by kaueltzen** [comment]
Thank you for your quick replies! I added a test both for skipping the relative stoichiometry check and appending a warning in the case of missing formula keys. There was already a test file in pymatgen, Li10GeP2S12.cif, that matched the criteria (it has no formula keys and sites with different multiplicities and occupancies).

**Review by janosh — APPROVED** [review]
very nice work! thanks a lot @kaueltzen 👍

**Pull request merged**